### PR TITLE
Update with_foundry

### PR DESCRIPTION
**Pull request by critesjosh**
# Description

Updates the with_foundry example to use 0.31.0

## Problem\*

Resolves 

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by critesjosh** [comment]
I still need to update testing workflow to use `bb` to generate the verifier contract

File: with-foundry/README.md [hunk 1]
```diff
@@ -24,12 +24,27 @@ Install [noirup](https://noir-lang.org/docs/getting_started/installation/#instal
...
+   proving backend for Noir (bb).
+
+   ```bash
+   curl -L https://raw.githubusercontent.com/AztecProtocol/aztec-packages/master/barretenberg/cpp/installation/install | bash
+   ```
+
+   then
+
+   ```bash
+   bbup -v 0.46.1 # compatible with nargo 0.31.0
```
Comment (Savio-Sou) [inline with-foundry/README.md:45]:
```suggestion
   bbup -v 0.41.0 # compatible with nargo 0.31.0
```

For Noir 0.31.0, the compatible bb version is 0.41.0

Comment (critesjosh) [reply with-foundry/README.md]: good catch

**Review by Savio-Sou — APPROVED** [review]
Minor comment but LGTM

**Pull request merged**